### ISSUE TITLE
main is red on build-test: setup-go pin drifted from hardened contract test

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#91]
## What

`main`'s `build-test` check (go-ci.yml) has been failing since commit 998cea5 (PR #88, "bump actions/setup-go from 6.5.0 to 7.0.0 in the github-actions group", merged 2026-07-22).

## Why

`internal/workflowcontract/wiki_docs_sync_test.go` hardcodes an expected exact pin for `actions/setup-go` in `wiki-docs-sync.yml`:

```
uses: actions/setup-go@924ae3a1cded613372ab5595356fb5720e22ba16 # v6.5.0
```

Dependabot's grouped `github-actions` update bumped `actions/setup-go` to `7.0.0` across workflow files, including `wiki-docs-sync.yml`, but this hardcoded contract test was not updated to match. Two tests now fail on every run:

- `TestWikiDocsSyncUsesHardenedGitHubAppContract`
- `TestWikiDocsSyncRejectsTriggerIdentityActionAndVerificationMutations/setup_Go_action_mutable_ref`

Both fail with the same root cause: the live workflow file's pin no longer matches the string the test expects.

## Impact

`main` has been red on `build-test` since 2026-07-22. This was found incidentally while verifying it would be safe to mark `build-test` a required status check (z-shell/zsh-lint#90); it is an independent, pre-existing defect and is being reported separately rather than fixed as part of that work.

## Suggested fix

Either update the hardcoded pin in `wiki_docs_sync_test.go` to the new reviewed `actions/setup-go@<v7 sha>` value, or revert `wiki-docs-sync.yml`'s `actions/setup-go` pin to `6.5.0` if this workflow is meant to require manual (non-Dependabot) review for that specific action. The test's name ("hardened contract") suggests the latter may be the intended design, in which case Renovate/Dependabot may need an ignore rule for this specific line rather than a one-time test update.

## References

- Failing run: https://github.com/z-shell/zsh-lint/actions/runs/29967538534
- Introduced by: z-shell/zsh-lint#88
- Found while investigating z-shell/zsh-lint#90
